src/Panels/AbstractPanel.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+namespace Wearesho\GoogleAutocomplete\Yii\Panels;
+
+use Wearesho\GoogleAutocomplete;
+use Wearesho\Yii\Http;
+
+/**
+ * Class AbstractPanel
+ * @package Wearesho\GoogleAutocomplete\Yii\Panels
+ */
+abstract class AbstractPanel extends Http\Panel
+{
+    /** @var string */
+    public $token;
+
+    /** @var string */
+    public $input;
+
+    /** @var GoogleAutocomplete\Enums\SearchLanguage */
+    public $language;
+
+    /** @var GoogleAutocomplete\ServiceInterface */
+    protected $service;
+
+    public function __construct(
+        GoogleAutocomplete\ServiceInterface $service,
+        Http\Request $request,
+        Http\Response $response,
+        array $config = []
+    ) {
+        parent::__construct($request, $response, $config);
+
+        $this->service = $service;
+    }
+
+    public function behaviors(): array
+    {
+        return [
+            'getBaseParams' => [
+                'class' => Http\Behaviors\GetParamsBehavior::class,
+                'attributes' => ['token', 'input',],
+            ],
+        ];
+    }
+
+    public function rules(): array
+    {
+        return [
+            [
+                ['token', 'input',],
+                'required',
+            ],
+            [
+                ['token', 'input', 'language'],
+                'string',
+            ],
+            ['language', 'string', 'length' => 2,],
+            ['language', 'default', 'value' => \Yii::$app->language,],
+            ['language', 'filter', 'filter' => function (string $language): GoogleAutocomplete\Enums\SearchLanguage {
```

Comment:
filter validator have not be used to transform scalar values to object values, because validation fill fail in case:
```php
<?php
$model = new \yii\base\Model;
$model->validate(); // true
$model->validate(); // TypeError: expected string, got SearchLanguage
```